### PR TITLE
Improvements for Bluewin/Swisscom TV

### PR DESCRIPTION
**Pull request by benib**
Relations:
  - Issue: https://github.com/livingdocsIO/livingdocs-planning/issues/3172
  - Related PRs:
    - https://github.com/livingdocsIO/q-editor/pull/2
    - https://github.com/livingdocsIO/Q/pull/2

## Description
This PR allows the editor config to be depending on the logged in user

#### Editor Plugin

- optionally take a Confidence style object as the editor config and send the session.credentials as criteria to the config getter

### CODE REVIEW

File: plugins/core/editor/routes/locales.js [hunk 1]
```diff
@@ -51,25 +60,35 @@ module.exports = {
...
         }
 
-        return translations;
+        return translations
       }
-    };
+    }
   }
-};
+}
```
Comment (marcbachmann) [inline plugins/core/editor/routes/locales.js]: A separate commit with the linting-specific changes would be good. That makes it easier to review the code.

File: plugins/core/editor/routes/locales.js [hunk 2]
```diff
@@ -51,25 +60,35 @@ module.exports = {
...
         // compute a translation.json file for use by i18next for the given language
-        let translations = {};
+        const translations = {}
+        let previewSizes
+
+        const criteria = request.auth ? request.auth.credentials : {};
+
+        // check for the get function to be backwards compatible if the config object gets
+        // passed directly instead of providing a Confidence getter
+        if (options.editorConfig.get instanceof Function) {
```
Comment (marcbachmann) [inline plugins/core/editor/routes/locales.js:70]: When do we plan to remove that check again? Often there's no need to make something backwards compatible.

File: plugins/core/editor/routes/locales.js (hunk 1 above)
Comment (benib) [reply plugins/core/editor/routes/locales.js]: done, I kept the style from before to not diverge the code style from the rest of the Q server code base.

File: plugins/core/editor/routes/locales.js (hunk 2 above)
Comment (benib) [reply plugins/core/editor/routes/locales.js:70]: I haven't planned the removal. We could as well do it bc breaking now. Would you prefer that?

**Review by marcbachmann — APPROVED** [review]

**Pull request merged**